### PR TITLE
Update kubedns and nodelocaldns to v1.22.28

### PR DESCRIPTION
**Pull request by mzaian**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Update kubedns and nodelocaldns versions. See release: [1.22.28](https://github.com/kubernetes/dns/releases/tag/1.22.28).

**Which issue(s) this PR fixes**:
Fixes NONE #

**Does this PR introduce a user-facing change?**:
NONE

```release-note
Update kubedns and nodelocaldns to release version 1.22.28
```

### CODE REVIEW

**Review by MrHohn — APPROVED** [review]
/lgtm
/approve
cc @DamianSawicki

**Comment by MrHohn** [comment]
Actually, holding to check if we need to do manual promotion in https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-dns/images.yaml.
/hold

**Comment by MrHohn** [comment]
Looks like that was done in https://github.com/kubernetes/k8s.io/pull/6068.

https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/images/k8s-staging-dns/images.yaml is the new image manifest location.

/unhold

**Comment by DamianSawicki** [comment]
Shouldn't we make a release fot the latest tag https://github.com/kubernetes/dns/releases/tag/1.22.28 and bump to that?

**Comment by MrHohn** [comment]
> Shouldn't we make a release fot the latest tag https://github.com/kubernetes/dns/releases/tag/1.22.28 and bump to that?

Thanks, I missed that - @mzaian let's pick 1.22.28 instead as there are CVE fixes.

/hold

**Comment by DamianSawicki** [comment]
New release created https://github.com/kubernetes/dns/releases/tag/1.22.28. Some time is certainly needed for the images to build.

**Comment by mzaian** [comment]
Thanks a lot for your feedback. I created https://github.com/kubernetes/k8s.io/pull/6088 to promote the images. Once merged I will update this PR to point to 1.22.28

**Comment by mzaian** [comment]
https://github.com/kubernetes/k8s.io/pull/6088 Is merged now

/unhold

**Comment by DamianSawicki** [comment]
I think this is ready to be merged, thank you @mzaian!

**Comment by aojea** [comment]
> I think this is ready to be merged, thank you @mzaian!

we are in code freeze, until master branc opens again next week this can not merge

**Comment by pacoxu** [comment]
https://github.com/kubernetes/kubernetes/pull/121736 bump to v1.22.27.

/retest

**Comment by mzaian** [comment]
> #121736 bump to v1.22.27.
> 
> /retest

Rebase is done and resolved all the conflicts

File: cluster/addons/dns/kube-dns/kube-dns.yaml.sed
Comment (pacoxu) [inline cluster/addons/dns/kube-dns/kube-dns.yaml.sed:224]:
```suggestion
        image: registry.k8s.io/dns/k8s-dns-sidecar:1.22.28
```

**Comment by DamianSawicki** [comment]
Thanks @mzaian!

/lgtm

**Pull request merged**
